broker-core/src/main/java/io/zeebe/broker/workflow/processor/BpmnStepContext.java

```diff
@@ -71,10 +148,15 @@ public void raiseIncident(ErrorType errorType, String errorMessage) {
         .setErrorMessage(errorMessage);
 
     if (!record.getMetadata().hasIncidentKey()) {
-      streamWriter.writeNewCommand(IncidentIntent.CREATE, incidentCommand);
+      commandWriter.writeNewCommand(IncidentIntent.CREATE, incidentCommand);
     } else {
-      streamWriter.writeFollowUpEvent(
-          record.getMetadata().getIncidentKey(), IncidentIntent.RESOLVE_FAILED, incidentCommand);
+      // TODO: casting is ok for the moment; the problem is rather that we
```

Comment:
@ThorbenLindhauer do we have a issue for this, sounds like something we should work on?

Reply:
I added a reference to https://github.com/zeebe-io/zeebe/issues/1033